diffrax/_integrate.py

```diff
@@ -957,6 +1036,21 @@ def _allocate_output(subsaveat: SubSaveAt) -> SaveState:
         dense_infos = None
         dense_save_index = None
 
+    if event is not None:
+        tprevprev = tprev
+        _, _, dense_info_for_event, _, _ = solver.step(
+            terms,
+            tprev,
+            tnext,
+            y0,
+            args,
+            solver_state,
+            made_jump,
+        )
```

Comment:
Note that this will actually evaluate the function, I think essentially pointlessly (as it will be re-evaluated in the loop).
I think using `jax.eval_shape` or `eqx.filter_eval_shape` will allow you to get the structure/dtype/shape of `dense_info_for_event` without having to actually do any work at runtime. (And without having to compile `solver.step` an additional time during compilation time either, which is a nontrivial concern.)